src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -1,116 +1,528 @@
+  const [regionForValue, setRegionForValue] = useState<IRegion | null>(null);
+  const [selectCities, setSelectCities] = useState<ICity[] | null>();
+  const [isCitySelect, setIsCitySelect] = useState<boolean>(false);
+  const [newAuthorValues, setNewAuthorValues] = useState<INewAuthorValues>({
+    avatar: '',
+    firstName: '',
+    lastName: '',
+    regionId: 0,
+    cityId: 0,
+    bio: '',
+    email: '',
+    /* eslint-disable-next-line */
+    socialNetwork: new Array(5).fill(null),
+  });
+  const [errorFields, setErrorFields] = useState({
+    avatar: fields.avatarRequired,
+    lastName: fields.emptyField,
+    firstName: fields.emptyField,
+    regionId: fields.emptyField,
+    cityId: fields.emptyField,
+    email: fields.emptyField,
+    work: fields.emptyField,
+    bio: fields.emptyField,
+    facebook: '',
+    instagram: '',
+    youtube: '',
+    twitter: '',
+    linkedin: '',
+    socialNetwoks: fields.socialNetworksEmpty,
+  });
+  const [visitFields, setVisitFields] = useState({
+    lastName: false,
+    firstName: false,
+    region: false,
+    city: false,
+    email: false,
+    work: false,
+    bio: false,
+    facebook: false,
+    instagram: false,
+    youtube: false,
+    twitter: false,
+    linkedin: false,
+  });
+  const [toggleButton, setToggleButton] = useState(true);
+
+  const setRegionByCityId = async (id: number) => {
+    const regionResponse = await getRegionByCityId(id).then((res) => {
+      if (res.status === 200) {
+        return res.data;
+      }
+      return false;
+    });
+    if (regionResponse) {
+      setSelectRegions([regionResponse]);
+      setRegionForValue(regionResponse);
     }
-  }, [expertId]);
+  };
 
-  useEffect(() => {
-    fetchExpert();
-  }, [fetchExpert]);
+  const setCitiesByRegionId = async (id: number) => {
+    const citiesResponse = await getCitiesByRegionId(id).then((res) => {
+      if (res.status === 200) {
+        return res.data;
+      }
+      return false;
+    });
+    if (citiesResponse) {
+      setSelectCities(citiesResponse);
+    }
+  };
+
+  const validateInput = (
+    value: string,
+    inputIdentifier: string,
+    acceptedSymbols: string,
+    minSymbols: number,
+    maxSymbols: number,
+    regexp: RegExp,
+  ): boolean => {
+    if (value && !regexp.test(value)) {
+      setErrorFields({ ...errorFields, [inputIdentifier]: acceptedSymbols });
+      return false;
+    }
+    if (!value || value.length < minSymbols) {
+      setErrorFields({
+        ...errorFields,
+        [inputIdentifier]: `Необхідно ввести мінімум ${minSymbols} символів`,
+      });
+      return false;
+    }
+    if (value.length > maxSymbols) {
+      setErrorFields({
+        ...errorFields,
+        [inputIdentifier]: `Можливо ввести максимум ${maxSymbols} символів`,
+      });
+      return false;
+    }
+    setErrorFields({ ...errorFields, [inputIdentifier]: '' });
+    return true;
+  };
+
+  const inputNameChangeHandler = (
```

Comment:
I think we better group validation rules in separate config object:

```
const validation = {
  name: {
    min: 2,
    max: 30,
    allowed: fields.acceptedSymbols,
  },
  bio: {
    min: 3,
    max: 250,
   
  }
```


Also I'm wondering if we need regexp when we validate just specific characters, not combinations of them etc. So maybe it makes sense to configure string with allowed characters? And use it to both validation and for error message?